### PR TITLE
[CBRD-24485] Refactoring create_or_drop_index_helper()

### PR DESCRIPTION
**Pull request by ctshim**
http://jira.cubrid.org/browse/CBRD-24485

* create_or_drop_index_helper():  Rewrite with fewer arguments.
* smt_add_index() :   The body of the smt_add_index() function does not exist, but it is declared externally, so it is deleted.

### CODE REVIEW

File: src/query/execute_schema.c [hunk 1]
```diff
@@ -2774,47 +2757,49 @@ create_or_drop_index_helper (PARSER_CONTEXT * parser, const char *const constrai
...
+
+  char *attname_tmp = NULL;
+  if (do_index != DO_INDEX_CREATE)
     {
-      assert (comment->node_type == PT_VALUE);
-      comment_str = (char *) PT_VALUE_GET_BYTES (comment);
+      assert (constraint_name != NULL);
+      nnames = 0;
+      attnames = &attname_tmp;
+      attnames[0] = NULL;
```
Comment (beyondykk9) [inline src/query/execute_schema.c:2768]: it looks not necessary. it is already NULL.

File: src/query/execute_schema.c [hunk 2]
```diff
@@ -2774,47 +2757,49 @@ create_or_drop_index_helper (PARSER_CONTEXT * parser, const char *const constrai
...
+      assert (idx_info);
+      nnames = pt_length_of_list (idx_info->column_names);
+
+      if (nnames == 1 && idx_info->prefix_length)
 	{
-	  colname = n->info.name.original;
+	  if (idx_info->column_names->info.sort_spec.expr)
+	    {
+	      char const *colname = idx_info->column_names->info.sort_spec.expr->info.name.original;
+	      if (sm_att_unique_constrained (obj, colname) || sm_att_fk_constrained (obj, colname))
```
Comment (beyondykk9) [inline src/query/execute_schema.c]: missed to check if colname is NULL ?

File: src/query/execute_schema.c (hunk 1 above)
Comment (ctshim) [reply src/query/execute_schema.c:2768]: It is put in such a way that developers who read the code are clearly aware that the names[0] should be NULL.
Excluding "nnames = 0;" and "attnames [0] = NULL;" is the same, but we want to keep it as it is for readability.

File: src/query/execute_schema.c (hunk 2 above)
Comment (ctshim) [reply src/query/execute_schema.c]: Thank you, I missed it.

**Review by beyondykk9 — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

**Review by hyunikn — APPROVED** [review]

**Review by youngjinj — APPROVED** [review]

**Pull request merged**